src/broker/forwarding_table.rs

```diff
@@ -45,7 +45,7 @@ impl ForwardingTable {
 
     /// Receives messages from publishers and forwards them to subscribers
     pub fn poll(&self) {
-        let mut buf = vec![vec![]; 16];
+        let mut buf = vec![vec![]; 2];
```

Comment:
Why this size reduction?

Reply:
restored this - then process the enqueue's in chunks of 2 (topic followed by message contents)

Reply:
Still not sure this is necessary as what is enqueued/dequeued on the ring buffer is only ever the first buffer in the chain

Though not opposed to having a size of 2 here, just the justification doesn't match requirements

Reply:
spoke offline, adjusted `mapping`'s enqueue/dequeue fn signatures to handle the topic more gracefully. worth calling out this does lead to an additional `Vec` allocation in `enqueue` to prepend the topic length + data before writing the entire chain.

one option to avoid this would be for `mapping` to support a `write_chain_at` & `read_chain_at` that can write/read from a particular byte offset of the chain, then we could write each "subchain" (topic length, topic, msg) of the broader chain individually w/o this intermediate allocation

curious on your thoughts re: avoiding that allocation at the expense of a bit more complexity

Reply:
IMO fine to have additional complexity to avoid allocation